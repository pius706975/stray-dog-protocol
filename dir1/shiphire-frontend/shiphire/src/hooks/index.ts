export * from './useAuth';
export * from './useRenter';
export * from './useShip';
export * from './useUser';
export * from './useHelper';
export * from './useShipOwner';
export * from './useAdmin';
export * from './useMutationWithErrorHandler';
