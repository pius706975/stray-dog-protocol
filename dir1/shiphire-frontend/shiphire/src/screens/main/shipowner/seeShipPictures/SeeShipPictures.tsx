import React from 'react';
import { View, Text, Image } from 'react-native-ui-lib';
import { SeeShipPicturesProps } from '../../../../types';
import { ScreenLayout, TextInput } from '../../../../components';
import { FontFamily, FontSize } from '../../../../configs';
import { useTranslation } from 'react-i18next';

const SeeShipPictures: React.FC<SeeShipPicturesProps> = ({
    navigation,
    route,
}) => {
    const {
        transactionId,
        sailingStatus,
        beforeSailingPictures,
        afterSailingPictures,
    } = route.params;
    const { t } = useTranslation('saveship');
    console.log('SeeShipPictures', JSON.stringify(route.params, null, 2));
    return (
        <ScreenLayout
            testId="ShipOwnerSeeShipPicturesScreen"
            backgroundColor="light"
            padding={10}
            gap={10}>
            <View
                style={{
                    padding: 10,
                }}>
                <Text
                    style={{
                        fontFamily: FontFamily.regular,
                        fontSize: FontSize.md,
                    }}>
                    {t('SeeShipPictures.PicturesHaveBeenSubmit')}
                </Text>
            </View>

            <View
                style={{
                    flexDirection: 'row',
                    justifyContent: 'space-between',
                }}>
                <View style={{ flex: 1 }}>
                    <Text
                        style={{
                            fontFamily: FontFamily.regular,
                            fontSize: FontSize.md,
                            marginBottom: 10,
                            textAlign: 'center',
                        }}>
                        {t('SeeShipPictures.textBefore')}
                    </Text>
                    {beforeSailingPictures &&
                        beforeSailingPictures.map((image, index) => (
                            <View key={index}>
                                <Image
                                    source={{ uri: image.documentUrl }}
                                    style={{
                                        width: 150,
                                        height: 150,
                                        borderRadius: 100 / 2,
                                        marginBottom: 10,
                                        alignSelf: 'center',
                                    }}
                                />
                                {beforeSailingPictures &&
                                    beforeSailingPictures[index] && (
                                        <TextInput
                                            placeholder="Description"
                                            value={
                                                beforeSailingPictures[index]
                                                    .description
                                            }
                                            editable={false}
                                        />
                                    )}
                            </View>
                        ))}
                </View>
                <View style={{ flex: 1 }}>
                    <Text
                        style={{
                            fontFamily: FontFamily.regular,
                            fontSize: FontSize.md,
                            marginBottom: 10,
                            textAlign: 'center',
                        }}>
                        {t('SeeShipPictures.textAfter')}
                    </Text>
                    {afterSailingPictures &&
                        afterSailingPictures.map((image, index) => (
                            <View key={index}>
                                <Image
                                    source={{ uri: image.documentUrl }}
                                    style={{
                                        width: 150,
                                        height: 150,
                                        borderRadius: 100 / 2,
                                        marginBottom: 10,
                                        alignSelf: 'center',
                                    }}
                                />
                                {beforeSailingPictures &&
                                    beforeSailingPictures[index] && (
                                        <TextInput
                                            placeholder="Description"
                                            value={
                                                beforeSailingPictures[index]
                                                    .description
                                            }
                                            editable={false}
                                        />
                                    )}
                            </View>
                        ))}
                </View>
            </View>
        </ScreenLayout>
    );
};

export default SeeShipPictures;
