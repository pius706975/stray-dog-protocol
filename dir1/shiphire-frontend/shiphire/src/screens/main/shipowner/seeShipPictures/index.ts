export { default as SeeShipPictures } from './SeeShipPictures';
