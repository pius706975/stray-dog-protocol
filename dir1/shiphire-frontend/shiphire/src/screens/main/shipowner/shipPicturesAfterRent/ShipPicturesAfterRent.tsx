import React from 'react';
import { View, Text, Image, Modal } from 'react-native-ui-lib';
import { ShipPicturesAfterRentProps } from '../../../../types';
import {
    useSubmitShipPicturesAfterSailing,
    useSubmitShipPicturesBeforeSailing,
} from '../../../../hooks';
import ImagePicker, { ImageOrVideo } from 'react-native-image-crop-picker';
import { useDispatch } from 'react-redux';
import { modalSlice, progressIndicatorSlice } from '../../../../slices';
import mime from 'mime';
import { handleAxiosError } from '../../../../utils';
import { Button, ScreenLayout, TextInput } from '../../../../components';
import { FontFamily, FontSize, PlusIcon } from '../../../../configs';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { ConfirmationModal } from '../shipPictures/components';
import { useTranslation } from 'react-i18next';

const ShipPicturesAfterRent: React.FC<ShipPicturesAfterRentProps> = ({
    navigation,
    route,
}) => {
    const { transactionId, beforeSailingPictures } = route.params;
    const [selectedImages, setSelectedImages] = React.useState<ImageOrVideo[]>(
        [],
    );
    const { t } = useTranslation('saveship');
    const [descriptions, setDescriptions] = React.useState<string[]>([]);
    const dispatch = useDispatch();
    const { hideModal, showModal } = modalSlice.actions;
    const { hideProgressIndicator, showProgressIndicator } =
        progressIndicatorSlice.actions;
    const mutationSubmitShipPicturesAfterSailing =
        useSubmitShipPicturesAfterSailing();
    const [isConfirmationModalVisible, setConfirmationModalVisible] =
        React.useState(false);

    const placeholderImages = beforeSailingPictures
        ? Array(beforeSailingPictures.length).fill(null)
        : [];

    const openImage = (position: number) => {
        ImagePicker.openPicker({
            multiple: true,
            maxFiles: 10,
            width: 300,
            height: 400,
            cropping: true,
        })
            .then((images: ImageOrVideo[]) => {
                console.log('masuk');

                const updatedImages = [...selectedImages];
                const updatedDescriptions = [...descriptions];

                // Replace the images and descriptions at the specified position
                updatedImages[position] = images[0]; // Assuming the user can only select one image at a time
                updatedDescriptions[position] = '';

                setSelectedImages(updatedImages);
                setDescriptions(updatedDescriptions);
            })
            .catch(e => console.log(e));
    };

    const submitImages = async () => {
        console.log('Selected Images:', selectedImages);
        console.log('Descriptions:', descriptions);
        const formData = new FormData();
        dispatch(showProgressIndicator());
        formData.append('transactionId', transactionId);
        selectedImages.forEach((image, index) => {
            formData.append('files', {
                uri: image.path,
                type: mime.getType(image.path),
                name: `Image After ${
                    (beforeSailingPictures &&
                        beforeSailingPictures[index]?.description) ||
                    ''
                }`,
            });

            formData.append('descriptions', descriptions[index]);
        });

        console.log('formData', formData);

        mutationSubmitShipPicturesAfterSailing.mutate(formData, {
            onSuccess: () => {
                dispatch(hideProgressIndicator());
                dispatch(
                    showModal({
                        status: 'success',
                        text: 'Ship pictures after sailing submitted successfully',
                    }),
                );

                setTimeout(() => {
                    dispatch(hideModal());
                    navigation.goBack();
                }, 2000);
            },
            onError: err => {
                dispatch(hideProgressIndicator());
                handleAxiosError(err);
                dispatch(
                    showModal({
                        status: 'failed',
                        text: 'Error submitting ship pictures',
                    }),
                );

                setTimeout(() => {
                    dispatch(hideModal());
                }, 2000);
            },
        });
    };

    const isFormValid = () => {
        // Check if at least one image is selected
        if (selectedImages.length === 0) {
            return false;
        }

        // Check if all description fields have values

        // If both conditions are met, the form is considered valid
        return true;
    };

    const showConfirmationModal = () => {
        setConfirmationModalVisible(true);
    };

    const hideConfirmationModal = () => {
        setConfirmationModalVisible(false);
    };

    const confirmSubmit = () => {
        hideConfirmationModal();
        submitImages();
    };

    return (
        <ScreenLayout
            testId="ShipPicturesAfterRentScreen"
            backgroundColor="light"
            padding={10}
            gap={10}>
            <View
                style={{
                    padding: 10,
                }}>
                <Text
                    style={{
                        fontFamily: FontFamily.regular,
                        fontSize: FontSize.md,
                    }}>
                    {t('ShipPicturesAfterRent.PleaseSubmit')}
                </Text>
            </View>

            <View
                style={{
                    flexDirection: 'row',
                    justifyContent: 'space-between',
                }}>
                {/* Left side - Display beforeSailingPictures */}
                <View style={{ flex: 1 }}>
                    <Text
                        style={{
                            fontFamily: FontFamily.regular,
                            fontSize: FontSize.md,
                            marginBottom: 10,
                            textAlign: 'center',
                        }}>
                        {t('ShipPicturesAfterRent.textBefore')}
                    </Text>
                    {beforeSailingPictures &&
                        beforeSailingPictures.map((image, index) => (
                            <View key={index} testID={`beforeSailing-${index}`}>
                                <Image
                                    source={{ uri: image.documentUrl }}
                                    style={{
                                        width: 150,
                                        height: 150,
                                        borderRadius: 100 / 2,
                                        marginBottom: 10,
                                        alignSelf: 'center',
                                    }}
                                />
                                {beforeSailingPictures &&
                                    beforeSailingPictures[index] && (
                                        <TextInput
                                            testId={`Desc-${index}`}
                                            placeholder="Description"
                                            value={
                                                beforeSailingPictures[index]
                                                    .description
                                            }
                                            editable={false}
                                        />
                                    )}
                            </View>
                        ))}
                </View>
                {/* // Right side - Select afterSailingPictures and provide descriptions */}
                <View style={{ flex: 1 }}>
                    <Text
                        style={{
                            fontFamily: FontFamily.regular,
                            fontSize: FontSize.md,
                            marginBottom: 10,
                            textAlign: 'center',
                        }}>
                        {t('ShipPicturesAfterRent.textAfter')}
                    </Text>
                    {placeholderImages.map((_, index) => (
                        <View key={index}>
                            <TouchableOpacity
                                testID="afterButton"
                                style={{
                                    width: 150,
                                    height: 150,
                                    borderRadius: 100 / 2,
                                    marginBottom: 10,
                                    alignSelf: 'center',
                                }}
                                onPress={() => openImage(index)}>
                                {/* Display selected image if available, otherwise display placeholder */}
                                {selectedImages[index] ? (
                                    <Image
                                        source={{
                                            uri: selectedImages[index].path,
                                        }}
                                        style={{
                                            width: 150,
                                            height: 150,
                                            borderRadius: 100 / 2,
                                            marginBottom: 10,
                                        }}
                                    />
                                ) : (
                                    <Image
                                        source={require('../../../../../assets/images/Empty.jpg')}
                                        style={{
                                            width: 150,
                                            height: 150,
                                            borderRadius: 100 / 2,
                                        }}
                                    />
                                )}
                            </TouchableOpacity>
                            {/* Display description for the selected image */}
                            {beforeSailingPictures &&
                                beforeSailingPictures[index] && (
                                    <TextInput
                                        testId={`desc-${index}`}
                                        placeholder="Description"
                                        value={
                                            beforeSailingPictures[index]
                                                .description
                                        }
                                        editable={false}
                                    />
                                )}
                        </View>
                    ))}
                </View>
            </View>

            <Button
                testID="submitButton"
                title={t('ShipPicturesAfterRent.Submit')}
                onSubmit={showConfirmationModal}
                disable={!isFormValid()}
            />

            <ConfirmationModal
                isVisible={isConfirmationModalVisible}
                onConfirm={confirmSubmit}
                onCancel={hideConfirmationModal}
            />
        </ScreenLayout>
    );
};

export default ShipPicturesAfterRent;
