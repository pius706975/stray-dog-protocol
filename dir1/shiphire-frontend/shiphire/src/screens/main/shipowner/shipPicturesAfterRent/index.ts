export { default as ShipPicturesAfterRent } from './ShipPicturesAfterRent';
