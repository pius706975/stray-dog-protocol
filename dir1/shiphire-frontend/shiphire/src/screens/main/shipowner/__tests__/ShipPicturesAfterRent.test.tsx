import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { QueryClient, QueryClientProvider } from 'react-query';
import {
    MainOwnerStackParamList,
    MainScreenOwnerParamList,
} from '../../../../types';
import MockAdapter from 'axios-mock-adapter';
import httpRequest from '../../../../services/api';
import store from '../../../../store';
import { Provider } from 'react-redux';
import { NavigationContainer } from '@react-navigation/native';
import { SeeShipPictures } from '../seeShipPictures';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { ShipPicturesAfterRent } from '../shipPicturesAfterRent';
import { setMockImagePickerResolve } from '../../../../jest/setup';
import { useSubmitShipPicturesAfterSailing } from '../../../../hooks';

jest.mock('../../../../hooks');
jest.mock('react-native-image-crop-picker', () => ({
    openPicker: jest.fn(() =>
        Promise.resolve([
            {
                creationDate: '1522437259',
                cropRect: null,
                data: null,
                duration: null,
                exif: null,
                filename: 'IMG_0006.HEIC',
                height: 3024,
                localIdentifier: 'CC95F08C-88C3-4012-9D6D-64A413D254B3/L0/001',
                mime: 'image/heic',
                modificationDate: '1530944178',
                path: '/Users/tabrezakhlaque/Library/Developer/CoreSimulator/Devices/3A21CC65-FCBF-4B7B-BE2C-87E18F17D6DC/data/Containers/Data/Application/C34290DF-2011-480C-9076-2B3CCFA8CD17/tmp/react-native-image-crop-picker/B02EB928-5871-4616-BA8F-05EF93037EB1.jpg',
                size: 2808983,
                sourceURL:
                    'file:///Users/tabrezakhlaque/Library/Developer/CoreSimulator/Devices/3A21CC65-FCBF-4B7B-BE2C-87E18F17D6DC/data/Media/DCIM/100APPLE/IMG_0006.HEIC',
                width: 4032,
            },
        ]),
    ),
    types: {
        image: 'image/jpg',
    },
}));

const queryClient = new QueryClient();
const MainOwnerStack = createNativeStackNavigator<MainOwnerStackParamList>();

const mockAdapter = new MockAdapter(httpRequest);

const mockSubmitImages = () => {
    mockAdapter
        .onPost('/ship-owner/submit-ship-pictures-after-sailing')
        .reply(200, {
            message: 'success',
        });
    (useSubmitShipPicturesAfterSailing as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onSuccess } = options;
            const mockResponseData = {
                message: 'success',
            };
            onSuccess(mockResponseData);
        }),
    }));
};
const shipPicturesAfterRentComponent = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainOwnerStack.Navigator>
                    <MainOwnerStack.Screen
                        name="ShipPicturesAfterRent"
                        component={ShipPicturesAfterRent}
                        initialParams={{
                            transactionId: '123',
                            sailingStatus: 'afterSailing',
                            beforeSailingPictures: [
                                {
                                    documentName: 'Ships',
                                    documentUrl: 'http://ship.com',
                                    description: 'this are the ship pictures',
                                },
                            ],
                            afterSailingPictures: {}[0],
                        }}
                    />
                </MainOwnerStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

describe('Ship Pictures After Rent', () => {
    describe('Snapshot', () => {
        // it('should match snapshot', () => {
        //     const tree = render(shipPicturesAfterRentComponent).toJSON();
        //     expect(tree).toMatchSnapshot();
        // })
    });
    describe('render correctly', () => {
        it('should render correctly', () => {
            const { getByTestId } = render(shipPicturesAfterRentComponent);

            expect(getByTestId('ShipPicturesAfterRentScreen')).toBeDefined();
        });
        it('should get image description of before rent pictures', () => {
            const { getByTestId } = render(shipPicturesAfterRentComponent);

            const desc = getByTestId('beforeSailing-0');
            expect(desc).toBeTruthy();
        });
        it('should submit button', async () => {
            mockSubmitImages();
            const { getByTestId } = render(shipPicturesAfterRentComponent);

            const afterBtn = getByTestId('afterButton');

            fireEvent.press(afterBtn);
            await waitFor(() => {
                setMockImagePickerResolve(true);
            });
            const submitButton = getByTestId('submitButton');

            fireEvent.press(submitButton);

            fireEvent.press(getByTestId('confirmButton'));
        });
        it('should cancel submit button', async () => {
            const { getByTestId } = render(shipPicturesAfterRentComponent);

            const afterBtn = getByTestId('afterButton');
            const submitButton = getByTestId('submitButton');
            const modal = getByTestId('modalConfirm');

            expect(afterBtn).toBeTruthy();

            setMockImagePickerResolve(true);

            fireEvent.press(submitButton);

            expect(submitButton).toBeDefined();

            expect(getByTestId('modalConfirm')).toBeDefined();

            fireEvent.press(modal);

            modal.props.onRequestClose(true);
        });
    });
});
