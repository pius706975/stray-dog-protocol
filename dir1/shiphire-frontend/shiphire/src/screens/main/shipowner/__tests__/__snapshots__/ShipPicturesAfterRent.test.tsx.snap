// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ship Pictures After Rent Snapshot should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="ShipPicturesAfterRentScreen"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": 10,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": 10,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "padding": 10,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "color": "#20303C",
                          "writingDirection": "ltr",
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "fontFamily": "Kanit-Regular",
                          "fontSize": 18,
                        },
                      ]
                    }
                  >
                    ShipPicturesAfterRent.PleaseSubmit
                  </Text>
                </View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "color": "#20303C",
                            "writingDirection": "ltr",
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "fontFamily": "Kanit-Regular",
                            "fontSize": 18,
                            "marginBottom": 10,
                            "textAlign": "center",
                          },
                        ]
                      }
                    >
                      ShipPicturesAfterRent.textBefore
                    </Text>
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      <Image
                        accessibilityRole="image"
                        accessible={false}
                        assetGroup="icons"
                        onError={[Function]}
                        source={
                          {
                            "uri": "http://ship.com",
                          }
                        }
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            false,
                            {
                              "alignSelf": "center",
                              "borderRadius": 50,
                              "height": 150,
                              "marginBottom": 10,
                              "width": 150,
                            },
                            false,
                          ]
                        }
                      />
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "width": "100%",
                          }
                        }
                      >
                        <View>
                          <View
                            style={
                              {
                                "paddingHorizontal": 10,
                                "width": "100%",
                              }
                            }
                            testID="RNE__Input__view-wrapper"
                          >
                            <View
                              collapsable={false}
                              style={
                                {
                                  "alignItems": "center",
                                  "borderBottomWidth": 1,
                                  "borderColor": "#216dab",
                                  "borderRadius": 0,
                                  "borderWidth": 0,
                                  "flexDirection": "row",
                                  "transform": [
                                    {
                                      "translateX": 0,
                                    },
                                  ],
                                }
                              }
                            >
                              <TextInput
                                editable={false}
                                multiline={false}
                                placeholder="Description"
                                placeholderTextColor="#86939e"
                                pointerEvents="none"
                                style={
                                  {
                                    "color": "#4d5963",
                                    "flex": 1,
                                    "fontFamily": "Kanit-Regular",
                                    "fontSize": 18,
                                    "minHeight": 40,
                                  }
                                }
                                testID="RNE__Input__text-input"
                                underlineColorAndroid="transparent"
                                value="this are the ship pictures"
                              />
                            </View>
                            <Text
                              style={
                                {
                                  "color": "#ff190c",
                                  "fontSize": 12,
                                  "margin": 5,
                                }
                              }
                            />
                          </View>
                          <View
                            accessibilityState={
                              {
                                "busy": undefined,
                                "checked": undefined,
                                "disabled": undefined,
                                "expanded": undefined,
                                "selected": undefined,
                              }
                            }
                            accessibilityValue={
                              {
                                "max": undefined,
                                "min": undefined,
                                "now": undefined,
                                "text": undefined,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              {
                                "position": "absolute",
                                "right": 14,
                                "top": 15,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "color": "#20303C",
                            "writingDirection": "ltr",
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "fontFamily": "Kanit-Regular",
                            "fontSize": 18,
                            "marginBottom": 10,
                            "textAlign": "center",
                          },
                        ]
                      }
                    >
                      ShipPicturesAfterRent.textAfter
                    </Text>
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      <RNGestureHandlerButton
                        collapsable={false}
                        delayLongPress={600}
                        enabled={true}
                        exclusive={true}
                        handlerTag={1}
                        handlerType="NativeViewGestureHandler"
                        onGestureEvent={[Function]}
                        onGestureHandlerEvent={[Function]}
                        onGestureHandlerStateChange={[Function]}
                        onHandlerStateChange={[Function]}
                        rippleColor={0}
                        touchSoundDisabled={false}
                      >
                        <View
                          accessible={true}
                          collapsable={false}
                          style={
                            {
                              "alignSelf": "center",
                              "borderRadius": 50,
                              "height": 150,
                              "marginBottom": 10,
                              "opacity": 1,
                              "width": 150,
                            }
                          }
                        >
                          <Image
                            accessibilityRole="image"
                            accessible={false}
                            assetGroup="icons"
                            onError={[Function]}
                            source={
                              {
                                "testUri": "../../../assets/images/Empty.jpg",
                              }
                            }
                            style={
                              [
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                false,
                                {
                                  "borderRadius": 50,
                                  "height": 150,
                                  "width": 150,
                                },
                                false,
                              ]
                            }
                          />
                        </View>
                      </RNGestureHandlerButton>
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "width": "100%",
                          }
                        }
                      >
                        <View>
                          <View
                            style={
                              {
                                "paddingHorizontal": 10,
                                "width": "100%",
                              }
                            }
                            testID="RNE__Input__view-wrapper"
                          >
                            <View
                              collapsable={false}
                              style={
                                {
                                  "alignItems": "center",
                                  "borderBottomWidth": 1,
                                  "borderColor": "#216dab",
                                  "borderRadius": 0,
                                  "borderWidth": 0,
                                  "flexDirection": "row",
                                  "transform": [
                                    {
                                      "translateX": 0,
                                    },
                                  ],
                                }
                              }
                            >
                              <TextInput
                                editable={false}
                                multiline={false}
                                placeholder="Description"
                                placeholderTextColor="#86939e"
                                pointerEvents="none"
                                style={
                                  {
                                    "color": "#4d5963",
                                    "flex": 1,
                                    "fontFamily": "Kanit-Regular",
                                    "fontSize": 18,
                                    "minHeight": 40,
                                  }
                                }
                                testID="RNE__Input__text-input"
                                underlineColorAndroid="transparent"
                                value="this are the ship pictures"
                              />
                            </View>
                            <Text
                              style={
                                {
                                  "color": "#ff190c",
                                  "fontSize": 12,
                                  "margin": 5,
                                }
                              }
                            />
                          </View>
                          <View
                            accessibilityState={
                              {
                                "busy": undefined,
                                "checked": undefined,
                                "disabled": undefined,
                                "expanded": undefined,
                                "selected": undefined,
                              }
                            }
                            accessibilityValue={
                              {
                                "max": undefined,
                                "min": undefined,
                                "now": undefined,
                                "text": undefined,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              {
                                "position": "absolute",
                                "right": 14,
                                "top": 15,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": true,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#6499c4",
                      "borderRadius": 6,
                      "elevation": 2,
                      "flexDirection": "row",
                      "gap": 0,
                      "justifyContent": "center",
                      "paddingHorizontal": 10,
                      "paddingVertical": 6,
                    }
                  }
                >
                  <Text
                    numberOfLines={0}
                    style={
                      {
                        "borderBottomWidth": 0,
                        "borderColor": undefined,
                        "color": "#f0f6fb",
                        "fontFamily": "Kanit-Medium",
                        "fontSize": 18,
                        "lineHeight": undefined,
                        "textAlign": undefined,
                      }
                    }
                  >
                    ShipPicturesAfterRent.Submit
                  </Text>
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "position": "absolute",
                          "right": 20,
                        },
                      ]
                    }
                  />
                </View>
                <Modal
                  animationType="slide"
                  hardwareAccelerated={false}
                  modifiers={
                    {
                      "alignments": {},
                      "backgroundColor": undefined,
                      "borderRadius": undefined,
                      "color": "transparent",
                      "flexStyle": undefined,
                      "margins": {},
                      "paddings": {},
                      "positionStyle": undefined,
                      "typography": undefined,
                    }
                  }
                  onRequestClose={[Function]}
                  transparent={true}
                  visible={false}
                />
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="ShipPicturesAfterRent"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
